src/source/Ice/IceAgent.c

```diff
@@ -2299,6 +2331,13 @@ STATUS handleStunPacket(PIceAgent pIceAgent, PBYTE pBuffer, UINT32 bufferLen, PS
 
             CHK_WARN(transactionIdStoreHasId(pIceCandidatePair->pTransactionIdStore, pBuffer + STUN_PACKET_TRANSACTION_ID_OFFSET), retStatus,
                      "Dropping response packet because transaction id does not match");
+
+            // Update round trip time and responses received only for relay candidates.
+            if(pIceCandidatePair->local->iceCandidateType == ICE_CANDIDATE_TYPE_RELAYED) {
+                pIceAgent->rtcIceServerDiagnostics[pIceCandidatePair->local->iceServerIndex].totalResponsesReceived++;
+                CHK_STATUS(hashTableGet(pIceAgent->rtcIceServerDiagnostics[pIceCandidate->iceServerIndex].requestTimestampDiagnostics, checkSum, &requestSentTime));
```

Comment:
Same

Reply:
Replaced with if check. You are right.Would want to exit if getting diagnostics item fails